src/tests/ftest/nvme/enospace.py

```diff
@@ -47,35 +47,35 @@ def setUp(self):
         self.create_pool_max_size()
 
     def verify_enospace_log(self, log_file):
-        """ Function checking logs consistency.
+        """Function checking logs consistency.
 
         Function checking that only expected errors have occurred and the DER_NOSPACE errors have
         occurred.
 
         Args:
-            log_file (string): name prefix of the log files to check.
+            log_file (str): name prefix of the log files to check.
         """
         logfile_glob = log_file + r".*[0-9]"
         errors_count = get_errors_count(self.hostlist_clients, logfile_glob)
         for error in self.expected_errors:
             if error not in errors_count:
                 errors_count[error] = 0
 
-        unexpected_errors_count = {
-            key: val for key, val in errors_count.items() if key not in self.expected_errors
-        }
-        if len(unexpected_errors_count) > 0:
+        if errors_count[self.DER_NOSPACE] <= 0:
+            self.fail(
+                "Expected DER_NOSPACE (-1007) should be > 0: got={}"
+                .format(errors_count[self.DER_NOSPACE]))
+
+        if len(self.expected_errors) != len(errors_count):
```

Comment:
Do we need to add the 0 entries to `errors_count` in line 60-62? Do we need those lines?

Reply:
Yes, it is needed for the following reasons:
- The test at line 64 could fail on element not in dict
- The test at line 69 could succeed even if unexpected errors are registred some errors from expected_errors where not raised.
- The test at line 82 could fail on element not in dict

Reply:
I see. Thanks for the explanation.

Reply:
My pleasure :)